### PR TITLE
Fix #18244: Invention DragWindow's starting position is inconsistent

### LINKED ISSUE #18244: Invention DragWindow's starting position is inconsistent
### Operating System

Windows 10, 64-bit

### OpenRCT2 build

OpenRCT2, v0.4.2 - latest, 9fe568a on develop

### Describe the issue

As you can see from the below attachment, window position of the Drag is not consistent and feels unnatural.
 

### Area(s) with issue?

_No response_

### Steps to reproduce

1. Open the scenario editor and advance to Invention List Setup
2. Choose one of the item from the list and try grabbing it from different cursor positions.

### Attachments

193753670-27bba247-e2a5-434a-8cf3-bf635bed30f9

### PR DESCRIPTION
**Pull request by raacker**
DragWindow's initial pos and size should match to the current scroll size and the InventionEditorWindow's position.

before
before

after

after1
after2

### CODE REVIEW

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 1]
```diff
@@ -640,36 +635,18 @@ class InventionDragWindow final : public Window
...
         auto screenCoords = windowPos + ScreenCoordsXY{ 0, 2 };
 
-        auto [drawString, ft] = WindowEditorInventionsListPrepareName(_draggedItem, true);
-        DrawTextBasic(&dpi, screenCoords, drawString, ft, { COLOUR_BLACK | COLOUR_FLAG_OUTLINE });
+        DrawResearchItem(
+            dpi, _draggedItem, width, screenCoords, STR_WINDOW_COLOUR_2_STRINGID, { COLOUR_BLACK | COLOUR_FLAG_OUTLINE });
     }
 
-    void Init(ResearchItem& researchItem)
+    void Init(ResearchItem& researchItem, const ScreenCoordsXY& editorPos, int scrollSize)
```
Comment (Broxzier) [inline src/openrct2-ui/windows/EditorInventionsList.cpp:642]: I think `objectSelectionScrollWidth` or something like that is clearer than `scrollSize`.

Comment (raacker) [reply src/openrct2-ui/windows/EditorInventionsList.cpp]: Agreed. Just updated.

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 2]
```diff
@@ -94,27 +95,34 @@ static void ResearchRidesSetup()
...
+
+        // Draw group name
         ft.Add<StringId>(rideTypeName);
-        ft.Add<StringId>(stringId);
+        DrawTextEllipsised(&dpi, screenCoords, columnSplitOffset - 11, format, ft, textPaint);
+
+        // Draw vehicle name
+        ft.Add<StringId>(itemNameId);
+        DrawTextEllipsised(
+            &dpi, { screenCoords.x + columnSplitOffset, screenCoords.y }, columnSplitOffset - 11, format, ft, textPaint);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorInventionsList.cpp:118]:
```suggestion
        DrawTextEllipsised(
            &dpi, { screenCoords + ScreenCoordsXY { columnSplitOffset, 0 }, columnSplitOffset - 11, format, ft, textPaint);
```

File: src/openrct2-ui/windows/EditorInventionsList.cpp [hunk 3]
```diff
@@ -94,27 +95,34 @@ static void ResearchRidesSetup()
...
-        StringId rideTypeName = get_ride_naming(researchItem.baseRideType, get_ride_entry(researchItem.entryIndex)).Name;
+        const StringId rideTypeName = get_ride_naming(researchItem.baseRideType, get_ride_entry(researchItem.entryIndex)).Name;
+
+        // Draw group name
         ft.Add<StringId>(rideTypeName);
-        ft.Add<StringId>(stringId);
+        DrawTextEllipsised(&dpi, screenCoords, columnSplitOffset - 11, format, ft, textPaint);
+
+        // Draw vehicle name
+        ft.Add<StringId>(itemNameId);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/EditorInventionsList.cpp:116]: Are you sure the `ft` Formatter doesn't need to be reset here like it used to? It probably does.

Comment (raacker) [reply src/openrct2-ui/windows/EditorInventionsList.cpp:116]: Oh it's very true. Vehicle name was just being appended there. Good catch.

**Review by tupaschoal — COMMENTED** [review]
This looks ok @raacker . Can you make sure the code formatting check is passing and also add a changelog entry, please?

**Comment by raacker** [comment]
@tupaschoal  added changelogs with dot trailing and double checked code formatting! sorry I literally just dumped the change and ran away :disappointed_relieved:

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**